File: Module_01_Big_O/Topic_05_Quadratic_Time_O(n²)/Python/example_02.py
"""
Course title: AlgoMania
Instructor: Thiago Brito - Universo DEV
Example adapted by: Charles Fernandes de Souza
Date: October 01, 2024
"""
def bubble_sort(my_array):
    n = len(my_array)
    # First iteration over the list (O(n))
    for i in range(n):
        # Second iteration for comparisons (O(n - i - 1)) ≈ O(n)
        for j in range(0, n - i - 1):
            # Comparison and swapping elements if needed
            if my_array[j] > my_array[j + 1]:
                my_array[j], my_array[j + 1] = my_array[j + 1], my_array[j]  # Swap elements
    return my_array

my_array = [30, 40, 20, 10]
sorted_array = bubble_sort(my_array)
sorted_array

File: Module_01_Big_O/Topic_05_Quadratic_Time_O(n²)/Java/Example02/src/com/souza/charles/Example02.java
package com.souza.charles;
/**
 Course title: AlgoMania
 Instructor: Thiago Brito - Universo DEV
 Example adapted by: Charles Fernandes de Souza
 Date: October 8, 2024
 */
public class Example02 {
    public static int[] bubbleSort(int[] myArray) {
        int n = myArray.length;
        // First iteration over the array (O(n))
        for (int i = 0; i < n; i++) {
            // Second iteration for comparisons (O(n - i - 1)) ≈ O(n)
            for (int j = 0; j < n - i - 1; j++) {
                // Comparison and swapping elements if needed
                if (myArray[j] > myArray[j + 1]) {
                    // Swap elements
                    int temp = myArray[j];
                    myArray[j] = myArray[j + 1];
                    myArray[j + 1] = temp;
                }
            }
        }
        return myArray;
    }

    public static void main(String[] args) {
        int[] myArray = {30, 40, 20, 10};
        int[] sortedArray = bubbleSort(myArray);
        for (int item: sortedArray) {
            System.out.print(item + " ");
        }
    }
}